src/lib.rs

```diff
@@ -523,6 +524,22 @@ impl<'a, R: Read> Iterator for FilesMut<'a, R> {
     }
 }
 
+impl Clone for Header {
+    fn clone(&self) -> Header {
+        let mut header = Header::new();
+        unsafe { ptr::copy_nonoverlapping(self, &mut header as *mut Header, 1) };
+        header
+    }
+}
+
+impl PartialEq for Header {
+    fn eq(&self, other: &Header) -> bool {
+        self.as_bytes()[..] == other.as_bytes()[..]
```

Comment:
I think in theory this may want to discard fields like `_rest` and perhaps discard trailing bytes after the last 0 byte in fields like `name`, (e.g. how one might compare C strings). Perhaps this could do field-wise comparison via the accessors below?